File: app/src/main/java/com/xjm/xxd/dribbird/login/LoginActivityPresenter.java
package com.xjm.xxd.dribbird.login;

import android.support.annotation.StringRes;

import com.xjm.xxd.dribbird.account.TokenBean;
import com.xjm.xxd.dribbird.account.TokenManager;

/**
 * Created by queda on 2016/12/2.
 */

public class LoginActivityPresenter implements ILoginActivityPresenter {

    private LoginActivityView mView;

    @Override
    public void bindIView(LoginActivityView iView) {
        mView = iView;
    }

    @Override
    public void init() {
        mView.loadUrl(TokenManager.getOAuth2Url());
    }

    @Override
    public void showLoading(String msg) {
        mView.showLoading(msg);
    }

    @Override
    public void showLoading(@StringRes int strId) {
        mView.showLoading(strId);
    }

    @Override
    public void hideLoading() {
        mView.hideLoading();
    }

    @Override
    public void loginSuccess(TokenBean tokenBean) {

    }

    @Override
    public void loginFailed() {

    }

    @Override
    public void onDestroy() {

    }

}

File: app/src/main/java/com/xjm/xxd/dribbird/login/LoginActivityPresenter.kt
package com.xjm.xxd.dribbird.login

import android.support.annotation.StringRes

import com.xjm.xxd.dribbird.account.TokenBean
import com.xjm.xxd.framework.network.GsonManager
import com.xjm.xxd.framework.data.CommonStore

/**
 * Created by queda on 2016/12/2.
 */

class LoginActivityPresenter : LoginActivityContract.Presenter() {

    override fun showLoading(@StringRes strId: Int) {
        viewer?.showLoading(strId)
    }

    override fun hideLoading() {
        viewer?.hideLoading()
    }

    override fun loginSuccess(tokenBean: TokenBean) {
        CommonStore.currentOauthToken = tokenBean.accessToken ?: ""
        CommonStore.currentTokenBean = GsonManager.gson().toJson(tokenBean)
        viewer?.loginSuccess()
    }

    override fun loginFailed() {
        viewer?.loginFailed()
    }

}
